py/bitbox02/bitbox02/bitbox02/bitbox02.py

```diff
@@ -108,6 +115,19 @@ def device_info(self) -> Dict[str, Any]:
             "monotonic_increments_remaining": response.device_info.monotonic_increments_remaining,
         }
 
+    def check_firmware_version(self) -> bool:
+        """
+        Returns False if the bitbox02 version is not what the library expects.
+        The connection should already bail early in case of a version mismatch,
+        so False indicates that some expected functionality might be missing.
+        """
+        # pylint: disable=no-member
+        _, _, edition, _ = self.get_info()
+
+        if edition == BitBox02Edition.MULTI:
+            return bool(self.version == MIN_BITBOX02_MULTI_FIRMWARE_VERSION)
```

Comment:
Why does the comparisons here use `==`.

Reply:
It shouldn't be necessary to typecast to `bool`.

Reply:
Mypy seems to think that `==` returns any - not sure why. 
The idea of this check is to establish whether the library will be able to use the full feature set of the firmware, or just a subset of it. The check is only useful, if the minimum firmware version if incremented each time we add a new firmware feature to the library. It can then function as a reminder to the users of the library that the library may be outdated.

Reply:
This is not what we discussed last time, is it? Bumping firmware minor version should not affect the Electrum user at all, as nothing can break (compatible change by definition).

The user should be forced to upgrade if the version is lower than `MIN_BITBOX02_MULTI_FIRMWARE_VERSION`. 

You proposed to require an upgrade if the version is too low, and tell the user to upgrade the library/Electrum if the version is higher than the next major version (e.g. for the min version 6.1.2, anything >= 7.0.0 would give the warning). That sounded good. What changed?

Reply:
>  It shouldn't be necessary to typecast to bool.

:+1:

Reply:
I can remove this check, if you think it is useless. It's not supposed to force anything, the check in `check_version` does that. It could act as a reminder that a newer version of the python library, that can use more features of the firmware compared to the currently used one might be available, but it's a pretty weak suggestion.